snitun/client/client_peer.py

```diff
@@ -119,14 +121,33 @@ async def start(
         )
 
         # Task a process for pings/cleanups
-        self._loop.create_task(self._handler())
+        if self._run_task:
+            raise RuntimeError("SniTun connection already running")
+        self._run_task = self._loop.create_task(self._handler())
```

Comment:
https://docs.python.org/3/library/asyncio-task.html
> Important Save a reference to the result of this function, to avoid a task disappearing mid-execution. The event loop only keeps weak references to tasks. A task that isn’t referenced elsewhere may get garbage collected at any time, even before it’s done. For reliable “fire-and-forget” background tasks, gather them in a collection: